File: python/concatenation_of_array.py
class Solution:
    def getConcatenation(self, nums: list[int]) -> list[int]:
        n: int = len(nums)
        result: list[int] = [0] * 2 * n
        for i in range(len(nums)):
            result[i] = nums[i]
            result[i + n] = nums[i]

        return result


def test_getConcatenation_case_1():
    # Arrange
    nums: list[int] = [1, 2, 1]
    expected: list[int] = [1, 2, 1, 1, 2, 1]

    # Act
    solution = Solution()
    actual = solution.getConcatenation(nums)

    # Assert
    assert expected == actual

File: rust/src/code/concatenation_of_array.rs
struct Solution {}

impl Solution {
    pub fn get_concatenation(nums: Vec<i32>) -> Vec<i32> {
        let n: usize = nums.len();
        let mut result: Vec<i32> = vec![0; 2 * n];
        for i in 0..nums.len() {
            result[i] = nums[i];
            result[i + n] = nums[i];
        }

        result
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn get_concatenation_case_1() {
        // arrange
        let nums: Vec<i32> = vec![1, 2, 1];
        let expected: Vec<i32> = vec![1, 2, 1, 1, 2, 1];

        // act
        let actual = Solution::get_concatenation(nums);

        // assert
        assert_eq!(expected, actual);
    }
}
